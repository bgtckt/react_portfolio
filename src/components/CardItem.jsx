import React from 'react';
import MyButton from "./UI/button/MyButton";

const CardItem = ({card, remove}) => {

  return (
    <div className='clients__card'>
      <h4 className='clients__name'>{card.id}. {card.name}</h4>
      <div className='clients__info'>
        <p><b>email:</b> {card.email}</p>
        <p><b>phone:</b> {card.phone}</p>
        <p><b>company:</b> "{card.company.name}"</p>
      </div>
      <MyButton onClick={() => remove(card)}>Удалить</MyButton>
    </div>
  );
};

export default CardItem;