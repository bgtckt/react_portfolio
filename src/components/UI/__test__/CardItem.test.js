import React from "react";
import {render, screen} from '@testing-library/react';
import CardItem from "../../CardItem";
import {BrowserRouter} from "react-router-dom";

const data = {
  id: 1,
  name: "Leanne Graham",
  email: "Sincere@april.biz",
  phone: "1-770-736-8031 x56442",
  company: {
    name: "Romaguera-Crona"
  }
};

describe('CardItem component', () => {
  it ('CardItem renders', () => {
    render(
      <BrowserRouter>
        <CardItem card={data}/>
      </BrowserRouter>
    );
    expect(screen.getByTestId('card-item')).toBeInTheDocument();
    expect(screen.getByText(/Leanne Graham/i)).toBeInTheDocument();
  });

  it ('CardItem snapshot', () => {
    const cardItem = render(
      <BrowserRouter>
        <CardItem card={data}/>
      </BrowserRouter>
    );
    expect(cardItem).toMatchSnapshot();
  });
});