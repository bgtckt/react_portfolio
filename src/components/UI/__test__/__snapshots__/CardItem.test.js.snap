// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardItem component CardItem snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="clients__card"
        data-testid="card-item"
      >
        <h4
          class="clients__name"
        >
          1
          . 
          Leanne Graham
        </h4>
        <div
          class="clients__info"
        >
          <p>
            <b>
              email:
            </b>
             
            Sincere@april.biz
          </p>
          <p>
            <b>
              phone:
            </b>
             
            1-770-736-8031 x56442
          </p>
          <p>
            <b>
              company:
            </b>
             "
            Romaguera-Crona
            "
          </p>
        </div>
        <button
          class="btn"
        >
          Удалить
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="clients__card"
      data-testid="card-item"
    >
      <h4
        class="clients__name"
      >
        1
        . 
        Leanne Graham
      </h4>
      <div
        class="clients__info"
      >
        <p>
          <b>
            email:
          </b>
           
          Sincere@april.biz
        </p>
        <p>
          <b>
            phone:
          </b>
           
          1-770-736-8031 x56442
        </p>
        <p>
          <b>
            company:
          </b>
           "
          Romaguera-Crona
          "
        </p>
      </div>
      <button
        class="btn"
      >
        Удалить
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
